zilliqa/src/consensus.rs

```diff
@@ -677,38 +679,103 @@ impl Consensus {
         Ok(None)
     }
 
-    fn get_next_committee(&mut self) -> Result<Committee> {
-        let mut committee = self.committee()?;
+    fn get_next_committee(&mut self, mut committee: Committee) -> Committee {
+        if committee.is_empty() {
+            panic!("committee is empty, this should never happen");
+        }
+
+        if !self.pending_peers.is_empty() {
+            info!(
+                "adding {} pending peers to committee",
+                self.pending_peers.len()
+            );
+        }
+
         committee.add_validators(self.pending_peers.drain(..));
-        Ok(committee)
+
+        if committee.len() <= 1 {
+            warn!(
+                "committee is too small {}, something might be wrong",
+                committee.len()
+            );
+        }
+
+        committee
+    }
+
+    fn are_we_leader_for_view(&mut self, parent_hash: Hash, view: u64) -> bool {
```

Comment:
I found the old way of getting the leader by using the view didn't handle when the views were not contiguous well, because it assumed there would be a block for view - 1 - which there isn't when a proposer has failed.